src/kademlia/get_item.cpp

```diff
@@ -170,92 +162,28 @@ bool get_item::invoke(observer_ptr o)
 
 void get_item::done()
 {
+	// only for put_immutable_item.
+	if (!m_data_callback) return find_data::done();
```

Comment:
is m_data_callback guaranteed to not be set for immutable puts? is it guaranteed to _be_ set for mutable puts?
it may be appropriate to put an assert here in either case (or expand the comment a bit)